ObjectMapper/Core/Mapper.swift

```diff
@@ -102,6 +102,13 @@ public final class Mapper<N: BaseMappable> {
 			}
 		}
 
+		if let klass = N.self as? ImmutableMappable.Type {
+			if let maybeObject = try? klass.init(map: map) as? N, var object = maybeObject {
```

Comment:
How about creating methods in the same scope with https://github.com/Hearst-DD/ObjectMapper/blob/master/ObjectMapper/Core/ImmutableMappble.swift#L144-L158? You can easily add `Mapper` methods for `ImmutableMappable`s.

The reason I defined mapping methods as extensions(#592) is to separate `ImmutableMappable` implementations from existing code.